core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -165,6 +165,14 @@ public static PartitionStatisticsFile computeAndWriteStatsFile(Table table, long
           "Using full compute as previous statistics file is not present for incremental compute.");
       stats = computeStats(table, snapshot, file -> true, false /* incremental */).values();
     } else {
+      if (statisticsFile.snapshotId() == snapshotId) {
+        // no-op
+        LOG.info(
+            "Snapshot {} already has statistics file computed. Returning the same file as result.",
```

Comment:
"Using existing statistics file for snapshot {}" probably suffices.

Reply:
updated.